addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -487,8 +484,20 @@ export class ClipboardPlugin extends Plugin {
             } else {
                 let childrenNodes;
                 if (node.nodeName === "DIV") {
-                    if (this.dependencies.baseContainer.isCandidateForBaseContainer(node)) {
-                        childrenNodes = childNodes(node);
+                    if (!node.hasChildNodes()) {
+                        node.remove();
+                        return;
+                    } else if (this.dependencies.baseContainer.isCandidateForBaseContainer(node)) {
+                        node.innerHTML = node.innerHTML.replace(/\n/g, "<br>");
```

Comment:
This does not seem correct. `\n` in a HTML content is collapsed by the browser and is visually a space, not a newline. Handling of `\n` is already done in `pasteText`.

What problematic use case are you trying to solve by adding this logic?
I don't see a test case for this.

@sbbh-odoo cc @dmo-odoo

Reply:
@sbbh-odoo This (your video) is standard behavior when pasting HTML content.

In Discord, they use the style:
```
white-space: break-spaces;
```
which allows newlines to be displayed as such in the browser.
Since that style is removed when pasting in a Odoo HTML, the text is displayed on a single line as expected.

In this case we should either specifically handle the `white-space` style when pasting, or do nothing at all in my opinion cc @dmo-odoo

Reply:
The issue I have with replacing `\n` with `<br>` is that we often have `\n` characters in our templates because of how we store them in `XML` files, I feel like it may introduce errors if we replace all `\n` indiscriminately by `<br>` elements.

One solution would be to detect that `white-space` style is present, then only do the conversion from `\n` to `<br>` at that point, and if the style is not present, not do it :)